### PR TITLE
Fix typos using codespell

### PR DESCRIPTION
**Pull request by hectormz**
# Description

This is a mostly automated typo-fix via [codespell](https://github.com/codespell-project/codespell/)

I looked through and fixed a few changes that were not really typos

## Type of change

- [X] Documentation/comments fix

# References

[codespell](https://github.com/codespell-project/codespell/)

### CODE REVIEW

File: napari/layers/shapes/_shape_list.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def remove(self, index, renumber=True):
             Location in list of the shape to be removed.
         renumber : bool
             Bool to indicate whether to renumber all shapes or not. If not the
-            expectation is that this shape is being immediately readded to the
+            expectation is that this shape is being immediately read to the
```
Comment (sofroniewn) [inline napari/layers/shapes/_shape_list.py]:
```suggestion
            expectation is that this shape is being immediately added back to the
```

**Review by sofroniewn — APPROVED** [review]
Wow this incredible @hectormz - everything here is great!! Very excited about this (as someone who spells badly!!)

We might want to mention this tool in our developer resources somewhere so it's easy to find and run again.

Curious what @tlambert03 and @jni think

**Comment by hectormz** [comment]
@sofroniewn It's overall pretty good! In addition to the mistake you found, there were perhaps two others that were wrong and ignored. Unfortunately it's not foolproof enough to add to CI or pre-commit. But documentation and occasional execution would be good

**Comment by jni** [comment]
> Curious what @tlambert03 and @jni think

@jni is a [grammar nazi](https://twitter.com/jnuneziglesias/status/1281193849528836096) and loves this! 😀 Thanks @hectormz!

**Comment by tlambert03** [comment]
Yes I also love this thanks!  Curious whether codespell can also find typos in non-comment code?  Like `BaseLayre`?

**Comment by sofroniewn** [comment]
> But documentation and occasional execution would be good

@hectormz given the enthusiasm can you make a one-line followup PR adding a link to the tool into this section of our developer resources https://github.com/napari/napari/blob/master/docs/source/developers/CORE_DEV_GUIDE.md#further-resources.

Something like:

[codespell](https://github.com/codespell-project/codespell/) for automated typo-fixing

We could also add to our developer dependencies if we wanted, https://github.com/napari/napari/blob/master/requirements/development.txt, though we might want to keep those from getting bloated, as this is strictly optional at this point. Curious what @jni and @tlambert03 think about that too?

**Comment by hectormz** [comment]
Sure, I can add that suggestion!

@tlambert03 from what I've seen with default settings, it doesn't look at non-comment code

**Pull request merged**